mmocr/models/textdet/postprocess/wrapper.py

```diff
@@ -391,3 +393,169 @@ def textsnake_decode(preds,
+
+    cls_pred = preds[0][0]
+    tr_pred = cls_pred[0:2].softmax(dim=0).data.cpu().numpy()
+    tcl_pred = cls_pred[2:].softmax(dim=0).data.cpu().numpy()
+
+    reg_pred = preds[1][0].permute(1, 2, 0).data.cpu().numpy()
+    x_pred = reg_pred[:, :, :2 * fourier_degree + 1]
+    y_pred = reg_pred[:, :, 2 * fourier_degree + 1:]
+
+    score_pred = tr_pred[1] * (tcl_pred[1]**tcl_tr_ratio)
+    tr_pred_mask = (score_pred) > tr_thresh
+    tr_mask = fill_hole(tr_pred_mask)
+
+    tr_contours, _ = cv2.findContours(
+        tr_mask.astype(np.uint8), cv2.RETR_TREE,
+        cv2.CHAIN_APPROX_SIMPLE)  # opencv4
+
+    mask = np.zeros_like(tr_mask)
+    e_matrix = generate_e_matrix(reconstr_points, fourier_degree)
+    boundaries = []
+    for cont in tr_contours:
+        deal_map = mask.copy().astype(np.int8)
+        cv2.drawContours(deal_map, [cont], -1, 1, -1)
+
+        text_map = score_pred * deal_map
+        polygons = contour_transfor_inv(fourier_degree, x_pred, y_pred, text_map, e_matrix,
+                                        scale)
+        polygons = poly_nms(polygons, nms_thresh)
+        boundaries = boundaries + polygons
+
+    boundaries = poly_nms(boundaries, nms_thresh)
+    return boundaries
+
+
+def poly_nms(polygons, threshold):
+    assert isinstance(polygons, list)
+
+    polygons = np.array(sorted(polygons, key=lambda x: x[-1]))
+
+    keep_poly = []
+    index = [i for i in range(polygons.shape[0])]
+
+    while len(index) > 0:
+        keep_poly.append(polygons[index[-1]].tolist())
+
+        A = Polygon(polygons[index[-1]][:-1].reshape(-1, 2)).buffer(0.01)
+
+        index = np.delete(index, -1)
+        iou_list = np.zeros((len(index), ))
+        for i in range(len(index)):
+            B = Polygon(polygons[index[i]][:-1].reshape(-1, 2)).buffer(0.01)
+
+            iou_list[i] = cal_iou(A, B)
+        remove_index = np.where(iou_list > threshold)
+        index = np.delete(index, remove_index)
+
+    return keep_poly
+
+
+def cal_iou(A, B):
+    if not A.intersects(B):
+        iou = 0
+    else:
+        area_AB = A.intersection(B).area
+        area_A = A.area
+        area_B = B.area
+        iou = area_AB / (area_A + area_B - area_AB)
+    return iou
+
+
+def contour_transfor_inv(fourier_degree, x_pred, y_pred, score_map, e_matrix,
+                         scale):
+    mask = score_map > 0
+
+    xy_text = np.argwhere(mask)
+    dxy = xy_text[:, 1] + xy_text[:, 0] * 1j
+
+    x = x_pred[mask]
+    y = y_pred[mask]
+
+    c = x + y * 1j
+    c[:, fourier_degree] = c[:, fourier_degree] + dxy
+    c *= scale
+
+    polygons = fourier_inverse_matrix(c, e_matrix=e_matrix)
+    score = score_map[mask].reshape(-1, 1)
+    return np.hstack((polygons, score)).tolist()
+
+
+def fourier_inverse_matrix(fourier_coeff, e_matrix):
+    ''' inverse fourier transform
+    Args:
+        fourier_coeff (np.ndarray): fourier coefficients shaped (n, 2k+1)
+        e_matrix (np.ndarray): a matrix of e^x, where x = 2pi x ikt and shape
```

Comment:
```suggestion
        exp_matrix (np.ndarray): a matrix of e^x, where x = 2pi x ikt and shape
```
what's ikt? pls cite the Equaiton in paper